### PR TITLE
[SPARK-29172][SQL] Fix some exception issue of explain commands

### PR DESCRIPTION
**Pull request by TomokoKomiyama**
### What changes were proposed in this pull request?
Added try exception

### Why are the changes needed?
The behaviors of run commands during exception handling are different depends on explain command. I think it should be unified.
[ >spark.sql("explain cost select * from hoge").show(false) ]
cost

[ >spark.sql("explain extended select * from hoge").show(false) ]
extemded

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
tested manually

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala [hunk 1]
```diff
@@ -156,7 +157,11 @@ case class ExplainCommand(
       sparkSession.sessionState.executePlan(logicalPlan))
     val outputString =
       if (codegen) {
+        try {
         codegenString(queryExecution.executedPlan)
+      } catch {
```
Comment (kiszk) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala]: Indentation should be fixed

Comment (TomokoKomiyama) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala]: Thank you for the review.
I've done.

**Review by srowen — COMMENTED** [review]
I'm not sure we want to let the explain plan proceed here? it has failed already, and I'm not sure about just appending it to the output as if successful.

**Comment by TomokoKomiyama** [comment]
@srowen 
If we use 'extend', sql behave 
```
scala> spark.sql("explain extended select * from hoge").show(false)
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|plan                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|== Parsed Logical Plan ==
'Project [*]
+- 'UnresolvedRelation [hoge]

== Analyzed Logical Plan ==
org.apache.spark.sql.AnalysisException: Table or view not found: hoge; line 1 pos 31;
'Project [*]
+- 'UnresolvedRelation [hoge]

org.apache.spark.sql.AnalysisException: Table or view not found: hoge; line 1 pos 31;
'Project [*]
+- 'UnresolvedRelation [hoge]

== Optimized Logical Plan ==
org.apache.spark.sql.AnalysisException: Table or view not found: hoge; line 1 pos 31;
'Project [*]
+- 'UnresolvedRelation [hoge]

== Physical Plan ==
org.apache.spark.sql.AnalysisException: Table or view not found: hoge; line 1 pos 31;
'Project [*]
+- 'UnresolvedRelation [hoge]
|
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```
, because of  the try exception in `sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala`(L156-)
```
    } catch {
      case e: AnalysisException => e.toString
    }
    append(analyzedOutput)
    append("\n")
    QueryPlan.append(analyzed, append, verbose, addSuffix, maxFields)
    append("\n== Optimized Logical Plan ==\n")
    QueryPlan.append(optimizedPlan, append, verbose, addSuffix, maxFields)
    append("\n== Physical Plan ==\n")
    QueryPlan.append(executedPlan, append, verbose, addSuffix, maxFields)
  }
```
Other explain commands doesn't behave like this, so it need something change to be consistent.
Can we add try exceptions in other explain commands to make these behaviors like 'extend' one?

**Comment by srowen** [comment]
OK, that's a good argument from consistency. @kiszk do you agree?

**Comment by kiszk** [comment]
I agree to make their behaviors in `explain` consistent.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**